### PR TITLE
Big refactoring: only git, Py 3.6-3.10, no buildout, GHA

### PR DESCRIPTION
**Pull request by mauritsvanrees**
This does most parts of issue #374.
Probably best to review this one commit at a time.
Note: I changed the `Dockerfile`, but did not test it. It might also no longer be needed.

Remaining from that issue, but not for this PR:

- Run `black` over the whole code. Are you okay with the defaults?
- Get rid of doctests. But this is likely a lot of work. BTW, the doctests may contain old code that `pyupgrade` did not touch and `black` will not touch, because it is in a text file.
- Use `pytest` instead of `zope.testrunner` and `z3c.testsetup`.

### CODE REVIEW

**Comment by reinout** [comment]
Wow, I'm going to take "a" look :-)

**Comment by reinout** [comment]
Black defaults are fine, btw.

File: zest/releaser/tests/postrelease.txt [hunk 1]
```diff
@@ -40,7 +30,7 @@ testing hack in utils.py:
 
 Run the postrelease script:
 
-    >>> utils.test_answer_book.set_answers(['', ''])
+    >>> utils.test_answer_book.set_answers(['', '', 'n'])
```
Comment (reinout) [inline zest/releaser/tests/postrelease.txt:31]: I've seen the extra `n` showing up a couple of times. Where's that coming from?
Perhaps "push to the server" that you didn't have with svn?

....

Ah wait, I see it below :-)

File: doc/source/developing.rst [hunk 2]
```diff
@@ -115,8 +95,11 @@ If you worked on the documentation, we suggest you verify the markup
 and the result by building the documentation locally and view your
 results.
 
-For building the documentation simply run::
+For building the documentation for the moment you still need buildout (untested)::
```
Comment (reinout) [inline doc/source/developing.rst]: I'll put that on my TODO list to fix it.

File: doc/source/developing.rst [hunk 3]
```diff
@@ -95,12 +95,12 @@ If you worked on the documentation, we suggest you verify the markup
...
 results.
 
-For building the documentation for the moment you still need buildout (untested)::
+For building the documentation::
 
-    $ python3.8 -mvenv .
-    $ bin/pip install zc.buildout setuptools
-    $ bin/buildout install sphinx
-    $ bin/sphinx
+    $ python3.9 -mvenv .
```
Comment (reinout) [inline doc/source/developing.rst:100]: `-m venv` with a space? At least, that's what I normally do.

File: doc/source/developing.rst [hunk 4]
```diff
@@ -95,12 +95,12 @@ If you worked on the documentation, we suggest you verify the markup
...
+    $ bin/pip install sphinx sphinx_rtd_theme
+    $ bin/pip install -e .
+    $ bin/sphinx-build doc/source/ doc/build/
```
Comment (reinout) [inline doc/source/developing.rst:103]: Ah, you already fixed it.

**Review by reinout — APPROVED** [review]
Basically: yes, looks fine! Hurray.
I'd say: merge it, then we can do some 'black' stuff.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[tool.isort]
+force_single_line = true
```
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply pyproject.toml:2]: I will add `profile = "plone"`.
See https://pycqa.github.io/isort/docs/configuration/profiles.html

File: doc/source/developing.rst (hunk 3 above)
Comment (mauritsvanrees) [reply doc/source/developing.rst:100]: Done

**Comment by mauritsvanrees** [comment]
@reinout I have pushed some minor fixes to address your comments.

File: pyproject.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (reinout) [reply pyproject.toml]: Hey, I didn't know those profiles existed! Plone is fine.
Is there a way of telling black not to mind the possibly long import lines?

**Comment by reinout** [comment]
Party!

File: pyproject.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (mauritsvanrees) [reply pyproject.toml]: Probably easier to add `line_length = 88` in the isort configuration, so black can be run without options. But so far so good.

**Pull request merged**